src/Pages/Shared/ConfigMapSecret/ProtectedConfigMapSecretDetails.tsx

```diff
@@ -0,0 +1,331 @@
+import { useEffect, useRef, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import Tippy from '@tippyjs/react'
+
+import {
+    DEPLOYMENT_HISTORY_CONFIGURATION_LIST_MAP,
+    DeploymentHistoryDiffView,
+    Progressing,
+    useAsync,
+    useUserEmail,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { ReactComponent as InfoIconOutlined } from '@Icons/ic-info-outline-grey.svg'
+import { hasApproverAccess, importComponentFromFELibrary } from '@Components/common'
+import { prepareHistoryData } from '@Components/app/details/cdDetails/service'
+import { DeploymentHistoryDetail } from '@Components/app/details/cdDetails/cd.type'
+import {
+    CMSecretComponentType,
+    CMSecretProtectedTab,
+    DraftState,
+    ProtectedConfigMapSecretProps,
+} from '@Pages/Shared/ConfigMapSecret/ConfigMapSecret.types'
+
+import { CM_SECRET_STATE } from './ConfigMapSecret.constants'
+import { ConfigMapSecretForm } from './ConfigMapSecretForm'
+import { getCMSecret } from './ConfigMapSecret.service'
+
+const ApproveRequestTippy = importComponentFromFELibrary('ApproveRequestTippy')
+
+export const ProtectedConfigMapSecretDetails = ({
+    appChartRef,
+    data,
+    id,
+    componentType,
+    cmSecretStateLabel,
+    isJobView,
+    selectedTab,
+    draftData,
+    parentName,
+    reloadEnvironments,
+    updateCMSecret,
+    openDeleteModal,
+    setOpenDeleteModal,
+}: ProtectedConfigMapSecretProps) => {
+    // HOOKS
+    const { appId, name } = useParams<{ appId: string; name: string }>()
+    const { email } = useUserEmail()
+
+    // STATES
+    const [baseData, setBaseData] = useState(data)
+
+    // REFS
+    const abortController = useRef(new AbortController())
+
+    // ASYNC CALLS
+    const [baseDataLoading, baseDataRes] = useAsync(
+        () =>
+            getCMSecret(componentType, id, appId, data.name, {
+                signal: abortController.current.signal,
+            }),
```

Comment:
- Either wrap with abortPreviousRequests as well or remove signal
- Get rid of the effect and move the logic to the service

Reply:
1. wrapped
2. data transformation in useEffect is not common to service, this is needed only for this file